pkg/api/controllers/dynamiclistener/listener_controller.go

```diff
@@ -98,7 +104,11 @@ func (c *Controller) enable(listener *v3.ListenConfig) error {
 
 func (c *Controller) updateCurrent(listener *v3.ListenConfig) error {
 	caCerts := strings.TrimSpace(listener.CACerts)
-	settings.CACerts.Set(caCerts)
+	toUpdate := false
+	if settings.CACerts.Get() != caCerts {
```

Comment:
after an offline discussion, we have decided to not only call update when the values are different, but also if the something had failed during the update in lines 137-158

Reply:
updated to place `toUpdate` outside `updateCurrent`, will be set to false on L158, only after the cluster sync logic successfully goes through